src/app/tests/suites/TestConstraints.yaml

```diff
@@ -265,3 +265,81 @@ tests:
       attribute: "char_string"
       arguments:
           value: ""
+
+    # Tests for hasValue attribute
+
+    - label: "Read attribute NULLABLE_INT8U Default Value"
+      command: "readAttribute"
+      attribute: "nullable_int8u"
+      response:
+          value: 0
+
+    - label: "Write attribute NULLABLE_INT8U with a value"
+      command: "writeAttribute"
+      attribute: "nullable_int8u"
+      arguments:
+          value: 0
+
+    - label: "Read attribute NULLABLE_INT8U with a value"
+      command: "readAttribute"
+      attribute: "nullable_int8u"
+      response:
+          constraints:
+              hasValue: true
+
+    - label: "Write attribute NULLABLE_INT8U without a value"
+      command: "writeAttribute"
+      attribute: "nullable_int8u"
+      arguments:
+          value: null
+
+    - label: "Read attribute NULLABLE_INT8U with a value"
+      command: "readAttribute"
+      attribute: "nullable_int8u"
+      response:
+          constraints:
+              hasValue: false
```

Comment:
But.... this is not in fact generating any code to check that there is no value, if you look at the generated code!

More generally, my understanding was that hasValue was about testing for value existence, not null vs not null?

Reply:
I had assumed python yaml parser is the one that validate these.

Reply:
Python right now can't tell apart missing and null @andy31415 so can't correctly implement hasValue anyway.